agent/engine/execcmd/manager_init_task_linux_test.go

```diff
@@ -452,3 +476,178 @@ func TestGetValidConfigExists(t *testing.T) {
+		task              *apitask.Task
+		expectedEndpoints expectedEndpoints
+		expectedError     string
+	}{
+		{
+			name:         "non-IPv6-only environment with bridge mode task",
+			sessionLimit: 5,
+			cfg: &config.Config{
+				InstanceIPCompatibility: ipcompatibility.NewIPv4OnlyCompatibility(),
+				AWSRegion:               "us-west-2",
+			},
+			task: bridgeTask,
+		},
+		{
+			name:         "IPv6-only environment with bridge mode task",
+			sessionLimit: 8,
+			cfg: &config.Config{
+				InstanceIPCompatibility: ipcompatibility.NewIPv6OnlyCompatibility(),
+				AWSRegion:               "us-east-1",
+			},
+			task:              bridgeTask,
+			expectedEndpoints: expectedDualstackEndpointsIAD,
+		},
+		{
+			name:         "IPv6-only task ENI with AWSVPC mode",
+			sessionLimit: 15,
+			cfg: &config.Config{
+				InstanceIPCompatibility: ipcompatibility.NewIPv4OnlyCompatibility(),
+				AWSRegion:               "us-east-1",
+			},
+			task:              ipv6OnlyTask,
+			expectedEndpoints: expectedDualstackEndpointsIAD,
+		},
+		{
+			name:         "Dual-stack task ENI with AWSVPC mode",
+			sessionLimit: 15,
+			cfg: &config.Config{
+				InstanceIPCompatibility: ipcompatibility.NewIPv6OnlyCompatibility(),
+				AWSRegion:               "us-east-1",
+			},
+			task: dualStackTask,
+		},
+		{
+			name:         "AWSVPC task without ENI",
+			sessionLimit: 5,
+			cfg: &config.Config{
+				InstanceIPCompatibility: ipcompatibility.NewIPv4OnlyCompatibility(),
+				AWSRegion:               "us-west-2",
+			},
+			task:          taskWithoutENI,
+			expectedError: "awsvpc mode task does not have a primary ENI",
+		},
+		{
+			name:         "Invalid region causes endpoint resolution failure",
+			sessionLimit: 5,
+			cfg: &config.Config{
+				InstanceIPCompatibility: ipcompatibility.NewIPv6OnlyCompatibility(),
+				AWSRegion:               "", // Empty region will cause endpoint resolution to fail
+			},
+			task:          bridgeTask,
+			expectedError: "failed to resolve SSM Messages endpoint: region is required to resolve ssmmessages dual stack endpoint",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result, err := formatSSMAgentConfig(tc.sessionLimit, tc.cfg, tc.task)
+
+			if tc.expectedError != "" {
+				assert.EqualError(t, err, tc.expectedError)
+				return
+			}
+
+			assert.NoError(t, err)
+
+			// Parse the JSON result into our struct
+			var config SSMAgentConfig
+			err = json.Unmarshal([]byte(result), &config)
+			require.NoError(t, err, "Result should be valid JSON")
+
+			// Check session workers limit is set correctly
+			assert.Equal(t, tc.sessionLimit, config.Mgs.SessionWorkersLimit, "Session workers limit should match")
+
+			// Verify container mode is true
+			assert.True(t, config.Agent.ContainerMode, "Agent.ContainerMode should be true")
+
+			// Assert each endpoint directly with its expected value from the test case
+			assert.Equal(t, tc.expectedEndpoints.mgs, config.Mgs.Endpoint)
```

Comment:
How does thie work for tests that did not have expectedEndpointsDefined/expectedErrors?
- non-IPv6-only environment with bridge mode task
- Dual-stack task ENI with AWSVPC mode